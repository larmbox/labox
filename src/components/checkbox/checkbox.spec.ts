import { mount } from '@vue/test-utils';
import { expect, test } from 'vitest';
import { LCheckbox } from '~/index';

test('mount component', async () => {
  expect(LCheckbox).toBeTruthy();

  const component = mount(LCheckbox, {
    props: {
      id: 'id',
    },
  });

  expect(component.html()).toMatchSnapshot();
  expect(component.html()).toContain(
    'form-control checkbox checkbox-md checkbox-primary'
  );
  expect(component.html()).not.toContain('form-field');
});

test('register css variables', async () => {
  mount(LCheckbox);
  expect(
    document.getElementById('lx-theme-provider-global')!.innerText
  ).toContain('--lx-checkbox');
});

test('inherit attributes', async () => {
  const component = mount(LCheckbox, {
    props: { notaprop: '1', variant: 'primary' },
  });
  expect(component.html()).toContain('notaprop="1"');
  expect(component.html()).not.toContain('variant="primary"');
});
