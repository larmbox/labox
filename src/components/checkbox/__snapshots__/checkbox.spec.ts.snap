// Vitest Snapshot v1

exports[`mount component 1`] = `
"<div class=\\"form-control checkbox checkbox-md checkbox-primary\\">
  <div class=\\"checkbox-inner\\">
    <div><input id=\\"id\\" type=\\"checkbox\\" />
      <div class=\\"checkbox-box\\"><span class=\\"icon check checkbox-icon\\" aria-hidden=\\"true\\"></span></div>
      <!--v-if-->
    </div>
    <div class=\\"checkbox-context\\">
      <!--v-if-->
      <!--v-if-->
    </div>
  </div>
</div>"
`;
